tests/ui/test_api.py

```diff
@@ -201,3 +203,64 @@ def test_ilsrecord(app, es_default_index, ils_record, ils_record_2):
     record.delete(delindex=True)
     assert len(RecordTest.get_all_pids()) == 0
     assert len(RecordTest.get_all_ids()) == 0
+
+
+class FailedPidIdentifier(RecordIdentifier):
+    """Sequence generator for Test identifiers."""
+
+    __tablename__ = 'failed_id'
+    __mapper_args__ = {'concrete': True}
+
+    recid = db.Column(
+        db.BigInteger().with_variant(db.Integer, 'sqlite'),
+        primary_key=True, autoincrement=True,
+    )
+
+
+FailedPidProvider = type(
+    'FailedPidProvider',
+    (Provider,),
+    dict(
+        identifier=FailedPidIdentifier,
+        pid_type='failed'
+    )
+)
+
+# cipo minter
```

Comment:
What is `cipo`

Reply:
Its comes from **ci**rculation **po**licies. My bad!